Source/WebKit/webpushd/PushClientConnection.mm

```diff
@@ -102,13 +107,32 @@
         if (!m_hostAppAuditToken)
             m_hostAppCodeSigningIdentifier = String();
         else
-            m_hostAppCodeSigningIdentifier = WebKit::codeSigningIdentifier(*m_hostAppAuditToken);
+            m_hostAppCodeSigningIdentifier = bundleIdentifierFromAuditToken(*m_hostAppAuditToken);
 #endif
     }
 
     return *m_hostAppCodeSigningIdentifier;
 }
 
+String ClientConnection::bundleIdentifierFromAuditToken(audit_token_t audit_token)
+{
+#if PLATFORM(MAC)
+    LSSessionID sessionID = (LSSessionID)audit_token_to_asid(audit_token);
+    auto auditTokenDataRef = adoptCF(CFDataCreate(kCFAllocatorDefault, (const UInt8 *)(&audit_token), sizeof(audit_token)));
+    CFTypeRef keys[] = { auditTokenDataRef.get() };
+    CFTypeRef values[] = { _kLSAuditTokenKey };
+    auto matchingAppsRef = adoptCF(_LSCopyMatchingApplicationsWithItems(sessionID, 1, keys, values));
+    if (matchingAppsRef && CFArrayGetCount(matchingAppsRef.get())) {
+        auto asnRef = adoptCF((LSASNRef)CFArrayGetValueAtIndex(matchingAppsRef.get(), 0));
```

Comment:
This looks like it will over-release. CFArrayGetValueAtIndex() doesn’t return a retained object.

Reply:
Great catch, I’ve got [a follow-up fix](https://github.com/WebKit/WebKit/pull/12686).